packages/protocol/contracts/L2/TaikoL2.sol

```diff
@@ -81,39 +85,53 @@ contract TaikoL2 is EssentialContract, IXchainSync {
      * @param l1Height The latest L1 block height when this block was proposed.
      * @param l1Hash The latest L1 block hash when this block was proposed.
      * @param l1SignalRoot The latest value of the L1 "signal service storage root".
+     * @param l2parentHash The expected parent hash.
      */
     function anchor(
         uint256 l1Height,
         bytes32 l1Hash,
-        bytes32 l1SignalRoot
+        bytes32 l1SignalRoot,
+        bytes32 l2parentHash
```

Comment:
@Brechtpd I realized in anchor transaciton, the current block's parent is not verified to be the same value as the parent hash in the block's header. With this extra input, the circuits can do the check.

Reply:
Alright. Checking that the previous block hash is correct is trivial because that blockhash is directly available in the circuit so it's just an equality check. It's all the other blockhashes that need to be calculated in the circuit that is the actual circuit work (which is why doing that in the smart contract still makes a lot of sense because that saves quite a bit or work on the circuit side).

Reply:
I ended up removing `l2parentHash` from anchor transaction input and emit an event that captures all block level variables mentioned in https://docs.soliditylang.org/en/v0.8.18/units-and-global-variables.html#block-and-transaction-properties

I'm assuming all these block level variables except `block.chainId`, and `blockhash(n-256)` ... `blockhash(n--2)` will be check by circuits using lookup table data.